File: Experiencia1.cpp
#include <iostream>

using namespace std;

class Figura {
public:
    virtual double calcularArea() = 0;
};

class Circulo : public Figura {
private:
    double radio;
public:
    Circulo(double r) : radio(r) {}
    double calcularArea() override {
        return 3.14 * radio * radio;
    }
};

class Rectangulo : public Figura {
private:
    double ancho, alto;
public:
    Rectangulo(double a, double b) : ancho(a), alto(b) {}
    double calcularArea() override {
        return ancho * alto;
    }
};

class Triangulo : public Figura {
private:
    double base, altura;
public:
    Triangulo(double b, double h) : base(b), altura(h) {}
    double calcularArea() override {
        return (base * altura) / 2;
    }
};

class Cuadrado : public Figura {
private:
    double lado;
public:
    Cuadrado(double l) : lado(l) {}
    double calcularArea() override {
        return lado * lado;
    }
};

class Pentagono : public Figura {
private:
    double lado, apotema;
public:
    Pentagono(double l, double a) : lado(l), apotema(a) {}
    double calcularArea() override {
        return (5 * lado * apotema) / 2;
    }
};

int main() {
    Figura* figura = nullptr;
    int opcion;

    cout << "Seleccione una figura:\n";
    cout << "1. Circulo\n";
    cout << "2. Rectangulo\n";
    cout << "3. Triangulo\n";
    cout << "4. Cuadrado\n";
    cout << "5. Pentagono\n";
    cout << "Opcion: ";
    cin >> opcion;

    switch(opcion) {
        case 1: {
            double radio;
            cout << "Ingrese el radio del circulo: ";
            cin >> radio;
            figura = new Circulo(radio);
            break;
        }
        case 2: {
            double ancho, alto;
            cout << "Ingrese el ancho del rectangulo: ";
            cin >> ancho;
            cout << "Ingrese el alto del rectangulo: ";
            cin >> alto;
            figura = new Rectangulo(ancho, alto);
            break;
        }
        case 3: {
            double base, altura;
            cout << "Ingrese la base del triangulo: ";
            cin >> base;
            cout << "Ingrese la altura del triangulo: ";
            cin >> altura;
            figura = new Triangulo(base, altura);
            break;
        }
        case 4: {
            double lado;
            cout << "Ingrese el lado del cuadrado: ";
            cin >> lado;
            figura = new Cuadrado(lado);
            break;
        }
        case 5: {
            double lado, apotema;
            cout << "Ingrese el lado del pentagono: ";
            cin >> lado;
            cout << "Ingrese la apotema del pentagono: ";
            cin >> apotema;
            figura = new Pentagono(lado, apotema);
            break;
        }
        default:
            cout << "Opcion no valida." << endl;
            return 1;
    }

    cout << "El area de la figura es: " << figura->calcularArea() << endl;

    delete figura;
    return 0;
}

File: Experiencia 1.py
class Figura:
    def calcular_area(self):
        pass

class Circulo(Figura):
    def __init__(self, radio):
        self.radio = radio

    def calcular_area(self):
        return 3.14 * self.radio * self.radio

class Rectangulo(Figura):
    def __init__(self, ancho, alto):
        self.ancho = ancho
        self.alto = alto

    def calcular_area(self):
        return self.ancho * self.alto

class Triangulo(Figura):
    def __init__(self, base, altura):
        self.base = base
        self.altura = altura

    def calcular_area(self):
        return (self.base * self.altura) / 2

class Cuadrado(Figura):
    def __init__(self, lado):
        self.lado = lado

    def calcular_area(self):
        return self.lado * self.lado

class Pentagono(Figura):
    def __init__(self, lado, apotema):
        self.lado = lado
        self.apotema = apotema

    def calcular_area(self):
        return (5 * self.lado * self.apotema) / 2

def main():
    opcion = int(input("Seleccione una figura:\n1. Circulo\n2. Rectangulo\n3. Triangulo\n4. Cuadrado\n5. Pentagono\nOpcion: "))

    if opcion == 1:
        radio = float(input("Ingrese el radio del circulo: "))
        figura = Circulo(radio)
    elif opcion == 2:
        ancho = float(input("Ingrese el ancho del rectangulo: "))
        alto = float(input("Ingrese el alto del rectangulo: "))
        figura = Rectangulo(ancho, alto)
    elif opcion == 3:
        base = float(input("Ingrese la base del triangulo: "))
        altura = float(input("Ingrese la altura del triangulo: "))
        figura = Triangulo(base, altura)
    elif opcion == 4:
        lado = float(input("Ingrese el lado del cuadrado: "))
        figura = Cuadrado(lado)
    elif opcion == 5:
        lado = float(input("Ingrese el lado del pentagono: "))
        apotema = float(input("Ingrese la apotema del pentagono: "))
        figura = Pentagono(lado, apotema)
    else:
        print("Opcion no valida.")
        return

    print("El area de la figura es: ", figura.calcular_area())

if __name__ == "__main__":
    main()
